### PR TITLE
Remove Decompiler.Setup Method

### PR DESCRIPTION
**Pull request by c3rb3ru5d3d53c**
When we pass a file object, we set the file's BINARY_ARCH and BINARY_MODE, which are abstractions of cs_arch and cs_mode. This will make the API much simpler to use as we only need to pass the file object in the constructor and not call Decompiler.Setup method.

### CODE REVIEW

File: include/common.h [hunk 1]
```diff
@@ -55,6 +55,19 @@ using std::setw;
...
 
+typedef enum BINARY_ARCH {
+    BINARY_ARCH_UNKNOWN = 0,
+	BINARY_ARCH_X86 = 1,
+	BINARY_ARCH_X86_64 = 2,
+} BINARY_ARCH;
+
+typedef enum BINARY_MODE {
+    BINARY_MODE_UNKNOWN = 0,
+	BINARY_MODE_32 = 1,
```
Comment (pisco-sour) [inline include/common.h:66]: Indentation off

Comment (c3rb3ru5d3d53c) [reply include/common.h]: Fixed :)

**Comment by c3rb3ru5d3d53c** [comment]
Alright so now there is no need to call `Decompiler.Setup`, so this should be removed, also in `ReadVector` for the file types, we don't need a `Setup` function here either, all the logic handling the now global `BINARY_` enums should be handled in `ReadVector`. As it will not be used anywhere else.

**Comment by c3rb3ru5d3d53c** [comment]
This code could be cleaner, we could opt for a `SetArchitecture` option with takes both the architecture and mode instead of using two functions for this. We can also move append queue into the `Decompile` method and have the `Decompile` method loop over the sections.

Before:
```c++
PE pe32;
pe32.SetBinaryArch(BINARY_ARCH_X86);
pe32.SetBinaryMode(BINARY_MODE_32);
if (pe32.ReadFile(g_args.options.input) == false){
    return EXIT_FAILURE;
}
Decompiler decompiler(pe32);
for (uint32_t i = 0; i < pe32.total_exec_sections; i++){
    decompiler.AppendQueue(pe32.sections[i].functions, DECOMPILER_OPERAND_TYPE_FUNCTION, i);
    decompiler.Decompile(pe32.sections[i].data, pe32.sections[i].size, pe32.sections[i].offset, i);
}
decompiler.WriteTraits();
return EXIT_SUCCESS;
```

```c++
PE pe32;
pe32.SetArchitecture(BINARY_ARCH_X86, BINARY_MODE_32);
if (pe32.ReadFile(g_args.options.input) == false){
    return EXIT_FAILURE;
}
Decompiler decompiler(pe32);
decompiler.Decompile();
decompiler.WriteTraits();
return EXIT_SUCCESS;
```

**Comment by c3rb3ru5d3d53c** [comment]
Some cleanup will need to be done on the CIL decompiler to remove the `Setup` method there as well.

There are some other todo items for bindings that will also need to be completed.

**Comment by c3rb3ru5d3d53c** [comment]
TODO: Fix raw processing

**Comment by c3rb3ru5d3d53c** [comment]
Alright I'm pretty happy with these changes now, will merge once the pipeline passes.

**Pull request merged**